src/_pytest/mark/structures.py

```diff
@@ -344,25 +345,22 @@ def with_args(self, *args: object, **kwargs: object) -> "MarkDecorator":
         mark = Mark(self.name, args, kwargs, _ispytest=True)
         return MarkDecorator(self.mark.combined_with(mark), _ispytest=True)
 
-    # Type ignored because the overloads overlap with an incompatible
-    # return type. Not much we can do about that. Thankfully mypy picks
-    # the first match so it works out even if we break the rules.
     @overload
-    def __call__(self, arg: Markable) -> Markable:  # type: ignore[misc]
+    def __call__(self, arg: Markable) -> Markable:
         pass
 
     @overload
-    def __call__(self, *args: object, **kwargs: object) -> "MarkDecorator":
+    def __call__(self, *args: Any, **kwargs: Any) -> "MarkDecorator":
```

Comment:
Any reason to change `object` to `Any` here?

Reply:
As mentioned above https://github.com/pytest-dev/pytest/pull/8674#discussion_r632996374, IIRC changing this to `Any` fixed how mypy was resolving the `@overloads` (but I can double check it wasn't the `Union` below)

Reply:
Yup, reverting this line only causes mypy to complain about the overlap again, I'm using:

mypy 0.812
Python 3.8.9

Reply:
To follow up more after reviewing the test file you linked to and the issues it links to, it looks like you've seen that using `object` is problematic for mypy (i.e. https://github.com/pytest-dev/pytest/pull/7496) and I'd probably suggest converting everything using `object` to `Any` to not try to convince `mypy` to associate a type since that's what you're saying when you use `object` (and I'm not sure why mypy doesn't treat `object` the same way it treats `Any`)

Reply:
OK, so using `Any` here is OK if it fixes some problem.